### ISSUE TITLE
[Suggestion] Custom Events and Types

### ISSUE THREAD

**Issue by Mwexim** [btk5h/skript-mirror#113]
**Is your feature request related to a problem? Please describe.**
It's not related to a problem, it's just an enhancement. I just want to create custom events, like custom syntax for conditions, effects and expressions.

**Describe the solution you'd like**
__For Events__
```
event player greet[ing]:
	variables:
		player
		greeting
	name: Greeting

on chat:
	if message contains "hi" or "hello":
		trigger event "Greeting" with event-player as player and with message as greeting.

# Then you could do
on player greet:
	# Do Stuff
```

__For Types__
```
type greeting with <(hi|hello|how are you)>:
	plural: greetings

# Then you can make an effect with this type
effect greet %player% with %greeting%:
	trigger:
		send expression-2 to expression-1

on chat:
	if message contains "hello custom bot":
		greet event-player with hi
```

**Additional context**
__For Events__
You just specify the event syntax like the others, but there are two extra things you can add:
- variable section
All the things listed here are used for e.g. event-player or event-item. 
- name section
Just the name for the trigger effect.
Then you create the actual event by using the trigger event syntax:
```trigger event %eventname% with %object% as %eventvariable%(,| and) with %object% as %eventvariable% ...```
You can have as many variables in your Events as you'd like.

__For Types__
I don't understand how you make Types in Java, so I made my own idea, you can improve my idea if you want.
So basically, you make a type with a regex.
```type %yourname% with <yourregex>```
Then you define the plural form
```plural: %pluralform%```

**Comment by Mwexim** [comment]
Closed because of inactivity.
